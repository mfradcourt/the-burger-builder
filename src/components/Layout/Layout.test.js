import React from 'react';
import { shallow } from "enzyme";

import Layout from './Layout';

describe('<Layout />', () => {
  let wrapper;

  it('should render Layout', () => {
    wrapper = shallow(
      <Layout>My Content</Layout>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should set showSideDrawer to false', () => {
    wrapper = shallow(
      <Layout>My Content</Layout>
    );
    const instance = wrapper.instance();
    instance.setState({'showSideDrawer': true});
    expect(wrapper.state('showSideDrawer')).toBe(true);
    instance.sideDrawerClosedHandler();
    expect(wrapper.state('showSideDrawer')).toBe(false);
  });

  it('should toggle the value of showSideDrawer', () => {
    wrapper = shallow(
      <Layout>My Content</Layout>
    );
    const instance = wrapper.instance();
    instance.setState({'showSideDrawer': true});
    expect(wrapper.state('showSideDrawer')).toBe(true);
    instance.sideDrawerToggleHandler();
    expect(wrapper.state('showSideDrawer')).toBe(false);
    instance.sideDrawerToggleHandler();
    expect(wrapper.state('showSideDrawer')).toBe(true);
  });
});
