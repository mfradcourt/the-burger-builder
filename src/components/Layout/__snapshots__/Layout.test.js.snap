// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render Layout 1`] = `
<Fragment>
  <toolbar
    drawerToggleClicked={[Function]}
  />
  <sideDrawer
    closed={[Function]}
    open={false}
  />
  <main
    className="Content"
  >
    My Content
  </main>
</Fragment>
`;
